### PR TITLE
cli: chmod 755 for backend plugin binaries

### LINKED ISSUE #15500: Grafana CLI should preserve permissions for backend binaries for Linux and Darwin
A new concept of backend plugins is being introduced to Grafana and one type of backend plugin will allow datasources to add alerting. These binary executables are defined in the plugin.json file for a plugin.

The grafana-cli currently resets all file permissions when extracting from a zip which results in the following error after installing a plugin with binaries and then restarting the Grafana server:

`t=2019-02-18T10:54:49+0000 lvl=eror msg="Failed to init plugin." logger=plugins error="fork/exec /var/lib/grafana/plugins/my-datasource/dist/my-plugin_linux_amd64: permission denied" plugin=my-datasource`

The original file permissions for the binary have been changed from `-rwxrwxr-x` to `-rw-rw-r--`.

The grafana-cli should preserve the original file permissions for plugin executables.

### PR DESCRIPTION
**Pull request by daniellee**
Fixes #15500. 

Does a simple filename check if the binary name ends with `_linux_amd64` or `_darwin_amd64` and if it does then sets the file mode to 755.

The zip file for the test includes 4 files that are all zero bytes in size (not real plugin binaries as they would be > 10 MB in size). See test for details on the file permissions for each file.

### CODE REVIEW

File: pkg/cmd/grafana-cli/commands/install_command_test.go [hunk 1]
```diff
@@ -37,3 +39,39 @@ func TestFoldernameReplacement(t *testing.T) {
 		})
 	})
 }
+
+func TestExtractFiles(t *testing.T) {
+	Convey("Should preserve file permissions for plugin backend binaries for linux and darwin", t, func() {
+		err := os.RemoveAll("./testfiles/fake-plugins-dir/grafana-simple-json-datasource")
```
Comment (bergquist) [inline pkg/cmd/grafana-cli/commands/install_command_test.go]: rename the folder `testdata` instead. 

> This trick is used in the standard library. I learned it from Mitchell Hashimoto’s and Dave Cheney’s work. go test has good support for loading test data from files. Firstly, go build ignores directory named testdata. Secondly, when go test runs, it sets current directory as package directory. This allows you to use relative path testdata directory as a place to load and store your data. Here is an example:

https://medium.com/@povilasve/go-advanced-tips-tricks-a872503ac859

**Review by bergquist — APPROVED** [review]
Let's Get The Money

The folder name change can be made in another commit if we want to merge this quickly.

**Comment by daniellee** [comment]
@bergquist No problem can rename the directory before merging.

File: pkg/cmd/grafana-cli/commands/install_command_test.go [hunk 2]
```diff
@@ -37,3 +39,39 @@ func TestFoldernameReplacement(t *testing.T) {
...
+		So(err, ShouldBeNil)
+
+		body, err := ioutil.ReadFile("./testfiles/grafana-simple-json-datasource-ec18fa4da8096a952608a7e4c7782b4260b41bcf.zip")
+		So(err, ShouldBeNil)
+
+		err = extractFiles(body, "grafana-simple-json-datasource", "./testfiles/fake-plugins-dir")
+		So(err, ShouldBeNil)
+
+		//File in zip has permissions 777
```
Comment (xlson) [inline pkg/cmd/grafana-cli/commands/install_command_test.go]: Comments are wrong. 755 not 777

Comment (xlson) [reply pkg/cmd/grafana-cli/commands/install_command_test.go]: I would probably go with: `Binary should be +rx to everyone.`

**Review by xlson — COMMENTED** [review]
Looks good! Test comments should be changed as to not confuse other devs, but that's it.

**Comment by xlson** [comment]
dont mind me and my inability to read code ;)

File: pkg/cmd/grafana-cli/commands/install_command_test.go (hunk 2 above)
Comment (daniellee) [reply pkg/cmd/grafana-cli/commands/install_command_test.go]: The original file is 777 so testing that it gets set to 755. The scenario is that when publishing plugins the permissions could be incorrect.

**Comment by daniellee** [comment]
Changed to use zf.Mode() instead so that it will work if someone uses a shell script to execute a binary (hopefully).

Have problems with the tests. Setting file mode 666 does not seem to work on CircleCI.

**Pull request merged**